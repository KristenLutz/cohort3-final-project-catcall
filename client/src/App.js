import React, { Component } from 'react';
import { connect } from 'react-redux';
// import apiService from './shared/services/api-service';

import './App.css';

import Input from './components/Input'
import Button from './components/Button'
import PageTitle from './components/PageTitle';

class App extends Component {
  // componentDidMount() {
  //     apiService
  //         .get('/someModels')
  //         .then(function (response) {
  //             console.log(response);
  //         })
  //         .catch(function (error) {
  //             console.log(error);
  //         });
  // }

  changeFilter = ev => this.props.dispatch({ type: 'SET_FILTER', payload: ev.target.value })

  startNewReport = () => { }

  viewReports = () => { }


  render() {
    const { filter } = this.props

    return (
      <div className="App">
        <PageTitle>CatCall.io</PageTitle>
        <Input value={filter} onChange={this.changeFilter} />
        <Button onClick={() => this.startNewReport}>Report Incident</Button>
        <Button onClick={() => this.viewReports}>View Reports</Button>
      </div>
    );
  }
}

const mapStateToProps = storeState => ({ filter: storeState.filter })
export default connect(mapStateToProps)(App)