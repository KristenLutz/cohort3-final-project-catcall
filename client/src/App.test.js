import React from 'react';
import renderer from 'react-test-renderer';
import ReactDOM from 'react-dom';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';

import createHistory from 'history/createBrowserHistory';
import { ConnectedRouter, routerMiddleware } from 'react-router-redux';

import App from './App';

it('renders without crashing', () => {
  const div = document.createElement('div');
  const history = createHistory();
  const middlewares = routerMiddleware(history);
  const mockStore = configureMockStore(middlewares);
  const initialState = {
    rootReducer: {
      locationReducer: {
        userInput: {},
        browserLocation: {},
      }
    }
  }
  const store = mockStore(initialState)

  ReactDOM.render(
    <Provider store={store}>
      <ConnectedRouter history={history}>
        <App />
      </ConnectedRouter>
    </Provider>, div
  );
  ReactDOM.unmountComponentAtNode(div);
});


it('renders correctly', () => {
  const history = createHistory();
  const middlewares = routerMiddleware(history);
  const mockStore = configureMockStore(middlewares);
  const initialState = {
    rootReducer: {
      locationReducer: {
        userInput: {},
        browserLocation: {},
      }
    }
  }
  const store = mockStore(initialState)
  const tree = renderer
    .create(<Provider store={store}>
      <ConnectedRouter history={history}>
        <App />
      </ConnectedRouter>
    </Provider>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
