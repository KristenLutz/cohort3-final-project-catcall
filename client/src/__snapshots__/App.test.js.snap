// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    className="App"
  >
    <div>
      <img
        alt="logo"
        src="/imgs/logo.png"
      />
      <div>
        <h1>
          CatCall.io
        </h1>
        <a
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </div>
    </div>
    <input
      onChange={[Function]}
      placeholder="Enter incident location"
      type="text"
      value="Loading location..."
    />
    <button
      onClick={undefined}
    >
      <a
        href="/report"
        onClick={[Function]}
      >
        Report Incident
      </a>
    </button>
    <button
      onClick={undefined}
    >
      <a
        href="/data"
        onClick={[Function]}
      >
        View Reports
      </a>
    </button>
  </div>
</div>
`;
